### PR TITLE
hdf5: specify the cmake_build_modules file correctly

### PR DESCRIPTION
**Pull request by paulharris**
Needed to add the line:
```
self.cpp_info.set_property("cmake_build_modules", [self._module_file_rel_path])
```

Otherwise the consumer doesn't see the generated cmake file.

### CODE REVIEW

File: recipes/hdf5/all/conanfile.py [hunk 1]
```diff
@@ -271,6 +271,7 @@ def _config_libname(lib):
         self.cpp_info.set_property("cmake_find_mode", "both")
         self.cpp_info.set_property("cmake_file_name", "HDF5")
         self.cpp_info.set_property("cmake_target_name", "HDF5::HDF5")
+        self.cpp_info.set_property("cmake_build_modules", [self._module_file_rel_path])
```
Comment (SpaceIm) [inline recipes/hdf5/all/conanfile.py:274]: Not like this please. You have to split the logic inside this file in 2 files:
- conan v1 tricks (target aliases) => to include into legacy cmake_find_package* generators
- `HDF5_IS_PARALLEL` variable => to include into legacy cmake_find_package* generators AND cmake_build_modules property (for CMakeDeps).

Comment (paulharris) [reply recipes/hdf5/all/conanfile.py]: ok, how about this next commit?

File: recipes/hdf5/all/conanfile.py [hunk 2]
```diff
@@ -261,16 +277,19 @@ def _config_libname(lib):
             self.cpp_info.components[component_name].libs = [_config_libname(alias_target)]
             self.cpp_info.components[component_name].requires = requirements
             self.cpp_info.components[component_name].includedirs.append(os.path.join("include", "hdf5"))
+            self.cpp_info.components[component_name].builddirs.append(os.path.join("lib", "cmake"))
```
Comment (SpaceIm) [inline recipes/hdf5/all/conanfile.py:280]:
```suggestion
```

File: recipes/hdf5/all/conanfile.py [hunk 3]
```diff
@@ -261,16 +277,19 @@ def _config_libname(lib):
...
-            self.cpp_info.components[component_name].build_modules["cmake_find_package"] = [self._module_file_rel_path]
-            self.cpp_info.components[component_name].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
+            self.cpp_info.components[component_name].build_modules["cmake_find_package"] = [self._module_targets_file_rel_path, self._module_variables_file_rel_path]
+            self.cpp_info.components[component_name].build_modules["cmake_find_package_multi"] = [self._module_targets_file_rel_path, self._module_variables_file_rel_path]
 
         self.cpp_info.set_property("cmake_find_mode", "both")
         self.cpp_info.set_property("cmake_file_name", "HDF5")
         self.cpp_info.set_property("cmake_target_name", "HDF5::HDF5")
+        self.cpp_info.set_property("cmake_build_modules", [self._module_variables_file_rel_path])
+        self.cpp_info.builddirs.append(os.path.join("lib", "cmake"))
```
Comment (SpaceIm) [inline recipes/hdf5/all/conanfile.py:292]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/hdf5/all/conanfile.py [hunk 4]
```diff
@@ -261,16 +277,19 @@ def _config_libname(lib):
...
             self.cpp_info.components[component_name].names["cmake_find_package_multi"] = component
-            self.cpp_info.components[component_name].build_modules["cmake_find_package"] = [self._module_file_rel_path]
-            self.cpp_info.components[component_name].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
+            self.cpp_info.components[component_name].build_modules["cmake_find_package"] = [self._module_targets_file_rel_path, self._module_variables_file_rel_path]
+            self.cpp_info.components[component_name].build_modules["cmake_find_package_multi"] = [self._module_targets_file_rel_path, self._module_variables_file_rel_path]
 
         self.cpp_info.set_property("cmake_find_mode", "both")
         self.cpp_info.set_property("cmake_file_name", "HDF5")
         self.cpp_info.set_property("cmake_target_name", "HDF5::HDF5")
+        self.cpp_info.set_property("cmake_build_modules", [self._module_variables_file_rel_path])
```
Comment (SpaceIm) [inline recipes/hdf5/all/conanfile.py:291]: are you sure it shouldn't be added at component level?

**Review by SpaceIm — APPROVED** [review]

File: recipes/hdf5/all/conanfile.py (hunk 4 above)
Comment (paulharris) [reply recipes/hdf5/all/conanfile.py]: Ok thanks

**Comment by paulharris** [comment]
@SpaceIm sorry, pls check latest commit, small tweak to try and fix conan-v2 pipeline error

**Review by SpaceIm — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**